src/api/views.py

```diff
@@ -116,3 +118,72 @@ def oai(request):
     }
 
     return render(request, template, context, content_type="application/xml")
+
+
+def kbart_tsv(request):
+    return kbart(request, delim='\t')
+
+
+def kbart(request, delim=','):
+    """
+    Produces KBART metadata output according to the spec:
+    https://doi.org/10.1080/0361526X.2017.1309826
+    @param request: the request object
+    @param delim: whether to use csv (',') or tsv ('\t') deliminators
+    @return: a rendered CSV or TSV
+    """
+    fields = ["date_first_issue_online", "num_first_vol_online",
+              "num_first_issue_online", "date_last_issue_online",
+              "num_last_vol_online", "num_last_issue_online",
+              "title_url", "first_author", "title_id", "embargo_info",
+              "coverage_depth", "notes", "publisher_name", "publication_type",
+              "date_monograph_published_print",
+              "date_monograph_published_online", "monograph_volume",
+              "monograph_edition", "first_editor",
+              "parent_publication_title_id", "preceding_publication_title_id",
+              "access_type"]
+
+    response = HttpResponse(content_type='text/csv')
+
+    writer = csv.DictWriter(response, fieldnames=fields, delimiter=delim)
+    response['Content-Disposition'] = 'attachment; filename="kbart.csv"' \
+        if delim == ',' else 'attachment; filename="kbart.tsv"'
+
+    writer.writeheader()
+
+    for journal in request.press.journals():
+        journal_line = {}
+        issues = journal.issues()
+        if issues.count() > 0:
+            journal_line['date_first_issue_online'] = \
+                '{:%Y-%m-%d}'.format(issues.order_by('date')[0].date)
+            journal_line['num_first_vol_online'] = \
+                issues.order_by('date')[0].volume
+            journal_line['num_first_issue_online'] = \
+                issues.order_by('date')[0].issue
+            journal_line['date_last_issue_online'] = \
+                '{:%Y-%m-%d}'.format(issues.order_by('-date')[0].date)
+            journal_line['num_last_vol_online'] = \
+                issues.order_by('-date')[0].volume
+            journal_line['num_last_issue_online'] = \
+                issues.order_by('-date')[0].issue
+
+        journal_line['title_url'] = journal.full_url()
```

Comment:
`journal.full_url()` is deprecated in favour of `journal.site_url()`

https://github.com/BirkbeckCTP/janeway/blob/cd9ebd1716e3a8592b5d021dd1fe994df449264e/src/journal/models.py#L237

Reply:
Changed